src/main/java/org/sagebionetworks/web/client/widget/table/modal/fileview/EntityViewScopeWidget.java

```diff
@@ -137,61 +121,20 @@ public void updateViewTypeMask() {
       );
   }
 
-  @Override
-  public void onSave() {
-    // update scope
-    synAlert.clear();
-    view.setLoading(true);
-    currentView.setScopeIds(editScopeWidget.getEntityIds());
-    currentView.setViewTypeMask(tableType.getViewTypeMask().longValue());
-    currentView.setType(null);
-    jsClient.updateEntity(
-      currentView,
-      null,
-      null,
-      new AsyncCallback<Entity>() {
-        @Override
-        public void onSuccess(Entity entity) {
-          view.setLoading(false);
-          view.hideModal();
-          eventBus.fireEvent(new EntityUpdatedEvent(entity.getId()));
-        }
-
-        @Override
-        public void onFailure(Throwable caught) {
-          view.setLoading(false);
-          synAlert.handleException(caught);
-        }
-      }
-    );
-  }
-
   @Override
   public void onEditScopeAndMask() {
-    List<Reference> references = getReferencesFromIdList(
-      currentView.getScopeIds()
+    String entityId = currentView.getId();
+    editEntityViewScopeModalWidget.configure(
+      entityId,
+      () -> {
+        editEntityViewScopeModalWidget.setOpen(false);
+        eventBus.fireEvent(new EntityUpdatedEvent(currentView.getId()));
+      },
+      () -> {
+        editEntityViewScopeModalWidget.setOpen(false);
+      },
+      true
```

Comment:
You don't need this if you change the signature as suggested above.

If you did keep this, then the `setOpen` below is redundant